langchain/chat_models/vertexai.py

```diff
@@ -129,8 +129,9 @@ def _generate(
         context = history.system_message.content if history.system_message else None
         params = {**self._default_params, **kwargs}
         if not self.is_codey_model:
-            params["context"] = context
-        chat = self.client.start_chat(**params)
+            chat = self.client.start_chat(context=context, **params)
```

Comment:
this is functionally equivalent, no? we're only adding "context" to params if it's not a codey model

Reply:
The start_chat logic is equivalent, but the error is caused by `chat.send_message(question.content, **params)`. In the previous version, context was added, but context isn't a valid argument for either session types (ChatSession and CodeChatSession). With this new change, context is never added to params

Here's the API reference for ChatSession.send_message: https://cloud.google.com/python/docs/reference/aiplatform/latest/vertexai.preview.language_models.ChatSession

Reply:
It's not, because you are using params variable in a different function, send_message just a few lines after and that's causing the issue. That function doesn't support context in params for any chat models for vertexai. Only chat-bison supports context at start chat function only.

Reply:
ah i see, thanks for explaining!